Automattic/jetpack
# SSO: Making sure SSO tooltip text is only added to the SSO Tooltip column

Fixes https://github.com/Automattic/jetpack/issues/36970

## Proposed changes:

* This PR fixes an issue where the SSO Tooltip text was displaying next to 'SSO Tooltip' within the Screen Options tab, on the View All Users page. To fix this, the text is inserted via JS.
* Note: As the SSO feature has now also been added to the connection package but the module itself is still used within the Jetpack plugin, this PR focusses purely on fixing this specific SSO tooltip issue rather than doing that as well as making sure the Jetpack plugin uses only the package version of SSO files. That is being done here: https://github.com/Automattic/jetpack/pull/37153
*  As such, the PHP changes have been made in both places (but the JS only in the package JS file which is referenced now in both places).

### Other information:


## Jetpack product discussion

https://github.com/Automattic/jetpack/issues/36970

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* To replicate the issue, on a test site with Jetpack connected, enable the SSO feature from the modules page at `/wp-admin/admin.php?page=jetpack_modules`.
* Navigate to Users > All Users.
* Click on 'Screen Options' (top right hand side of page) and you'll see the tooltip and text showing next to 'SSO Status'.

To test the fix:
* Check out this PR locally or using the Beta tester plugin, and follow the same steps as above. 
* When clicking 'Screen Options' you should only see the text 'SSO Status'.
* When hovering over the 'SSO Status' in the user table column, the tooltip text should show in the tooltip, (both at the top and bottom of the table depending on which text you hover over).
* Add a new user who does not have SSO enabled, and you should see another info icon next to 'send invite' in the SSO Status column for that user. Hover over that and you should see different info text (`This user doesn’t have an SSO connection to WordPress.com. Invite them to the site to increase security and improve their experience`) - this info text shouldn't be affected by the changes made in this PR, basically.
* Note that this code is only testing the PHP changes from the module file (JS changes are directly in the package file version), but the same code is used in the package file as well.


### Screenshots:

Expected SSO Status tooltip display:

<img width="371" alt="Screenshot 2024-05-07 at 16 26 00" src="https://github.com/Automattic/jetpack/assets/16754605/f9a2e6ed-e5dc-4366-9d3c-9ad6d305b865">


Expected info icon for non-SSO enabled user:
<img width="380" alt="Screenshot 2024-05-07 at 16 26 05" src="https://github.com/Automattic/jetpack/assets/16754605/313ef9ae-d34b-480a-9867-a26c741422af">


Expected screen options display:
<img width="1165" alt="Screenshot 2024-05-07 at 16 26 11" src="https://github.com/Automattic/jetpack/assets/16754605/6c1eba02-7fca-4589-b192-61eba2787f21">
